### PR TITLE
docs: rename pat to api key

### PR DESCRIPTION
**Pull request by tomekpapiernik**
This PR renames oryAccessToken to APIKey to make the API docs in line with Ory Console and ory.sh/docs.

BREAKING CHANGES: Renamed the `oryAccessToken` to `oryApiKey`. If you're using the `oryAccessToken` to interact with the API, please rename the parameter to `oryApiKey` as well.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

## Further Comments

Replaces the previous PR. https://github.com/ory/kratos/pull/3091

### CODE REVIEW

File: internal/client-go/api_courier.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func (a *CourierApiService) GetCourierMessageExecute(r CourierApiApiGetCourierMe
 	if r.ctx != nil {
 		// API Key Authentication
 		if auth, ok := r.ctx.Value(ContextAPIKeys).(map[string]APIKey); ok {
-			if apiKey, ok := auth["oryAccessToken"]; ok {
+			if apiKey, ok := auth["oryApiKey"]; ok {
```
Comment (zepatrik) [inline internal/client-go/api_courier.go:134]: Hm this does cause breaking changes :thinking: No idea tbh on how we can patch this to fall back to the old key if it is not provided...
I guess there is no way, breaking changes in the SDK are unavoidable with these sh... generator templates in general, but old SDK versions should still be compatible with the server.

**Review by zepatrik — APPROVED** [review]

File: internal/client-go/api_courier.go (hunk 1 above)
Comment (aeneasr) [reply internal/client-go/api_courier.go:134]: The question is whether we have to break this, just to update the naming of this key? Maybe we can fix this in all services in one go etc

Comment (aeneasr) [reply internal/client-go/api_courier.go:134]: This also means that we have to update all the docs for the Go examples btw

**Review by aeneasr — CHANGES_REQUESTED** [review]
I think more work is required to get this streamlined in also other servers, and then in the technical docs as well

**Pull request closed without merging**